src/rest_api/classes/person/widgets/laboratory.clj

```diff
@@ -0,0 +1,122 @@
+(ns rest-api.classes.person.widgets.laboratory
+  (:require
+   [datomic.api :as d]
+   [pseudoace.utils :as pace-utils]))
+
+
+(defn lab-info [person]
+   (let [db (d/entity-db person)
+         data
+         (->> (d/q '[:find [?laboratory ...]
+                   :in $ ?person
+                   :where [?laboratory :laboratory/registered-lab-members ?person]
+                          ]
+                 db (:db/id person))
+              (map (fn [oid]
+                     (let [laboratory (d/entity db oid)]
+                       (pace-utils/vmap
+                        :strain
+                        (pace-utils/vmap
+                         :taxonomy "all"
+                         :class "laboratory"
+                         :label (:laboratory/id laboratory)
+                         :id (:laboratory/id laboratory))
+                        :lab
+                        (pace-utils/vmap
+                         :taxonomy "all"
+                         :class "laboratory"
+                         :label (:laboratory/id laboratory)
+                         :id (:laboratory/id laboratory))
+                        :allele (:laboratory/allele-designation laboratory)
+                        :rep
+                        (->> (:laboratory/representative laboratory)
+                             (map (fn [rep]
+                                  {:taxonomy "all"
+                                   :class "person"
+                                   :label (:person/standard-name rep)
+                                   :id (:person/id rep)}) ))
+                       )
+                     )))
+              (seq))]
+     {:data (if (empty? data) nil data)
+   :description
+   "allele designation of the affiliated laboratory"}))
+;; note that the perl catalyst interface does not return proper data for "rep", it gives a single value while the data could have multiple persons, e.g. WBPerson8705
+
+(defn gene-classes [person]
```

Comment:
Could this be re-worked so just one datomic query is made?
Currently: 1 + number-of-labs-per-person queries.

Something like:
```clojure
(d/q '[:find [?geneclass ...]
         :where
         [?lab :laboratory/registered-lab-members ?person]
         [?geneclass :gene-class/designating-laboratory ?lab]]
        db (:db/id person))
```